policy.go

```diff
@@ -298,6 +380,54 @@ func newSampledLFU(maxCost int64) *sampledLFU {
 	}
 }
 
+func newSampledLFUFromSnapshot(file string) (*sampledLFU, error) {
+	var err error
+
+	f, err := os.Open(file)
+	if err != nil {
+		return nil, err
+	}
+
+	stat, err := f.Stat()
+	if err != nil {
+		return nil, err
+	}
+
+	buf, err := z.NewReadBuffer(f, int(stat.Size()))
+	if err != nil {
+		return nil, err
+	}
+
+	return UnmarshalSampledLFU(buf.Bytes())
+}
+
+func (p *sampledLFU) MarshalToBuffer(buffer io.Writer) error {
+	export := sampledLFUExport{
+		MaxCost:  p.maxCost,
+		Used:     p.used,
+		KeyCosts: p.keyCosts,
+	}
+
+	e := msgpack.NewEncoder(buffer)
```

Comment:
Good suggestion :+1: I'm taking a look now at how easy this will be. Part of the compromise I had in mind here was a solution that was minimally intrusive + easy to use from the client perspective - greenpack can definitely still satisfy that but just will add some codegen at the very least. From a perf perspective though minimizing allocs during the encoding step could be a big win w/ something like greenpack.

Reply:
Discussed offline, but after running some simple benchmarks I'm not sure it's worth it. Some results for a TinyLFU with 500k keys and 200 million counters:

```
BenchmarkMarshalTinyLFUGreen-10      	       1	15587347208 ns/op	4872098832 B/op	      40 allocs/op
BenchmarkMarshalTinyLFUMsgpack-10    	       1	32429659958 ns/op	4872091736 B/op	      30 allocs/op
```

The number of allocs and total allocated memory is more or less the same. Greenpack is a bit faster, but this is less important IMO and the degree here may not be worth the additional code/struct tagging needed

Reply:
It might not be as useful for TinyLFU, but how does it perform for the cache maps?